framework/framework.go

```diff
@@ -142,3 +142,60 @@ func (f *Framework) ProcessDelete(obj interface{}, resources []Resource) error {
 
 	return nil
 }
+
+// ProcessUpdate is a drop-in for an informer's UpdateFunc. It receives the
+// custom object observed during TPR watches and anything that implements
+// Resource. ProcessUpdate takes care about all necessary reconciliation logic
+// for update events. For complex resources this means state has to be created,
+// deleted and updated eventually, in this order.
+//
+//     func updateFunc(oldObj, newObj interface{}) {
+//         err := f.ProcessUpdate(oldObj, newObj, resources)
+//         if err != nil {
+//             // error handling here
+//         }
+//     }
+//
+//     newResourceEventHandler := &cache.ResourceEventHandlerFuncs{
+//         UpdateFunc:    updateFunc,
+//     }
+//
+func (f *Framework) ProcessUpdate(oldObj, newObj interface{}, resources []Resource) error {
```

Comment:
As I see it an operator needs two pieces of information to be able to reconcile. One is the current state and the other is the desired state. The current state has to be looked up using the "old" custom object, since this is the piece of information that lead to the current state in former iterations. This can only not work in case the old custom object is empty or wrong, which I cannot really prove at this point, but I would be very surprised by this. The old object is only be used in `GetCurrentState`. The rest of the methods receive the new value. I think it is correct that way. Can you elaborate on a specific problem that you see with this approach?